(Android) E/onesignal: Unable to initialize the OneSignal SDK because activity is null true or oneSignalInitDonefalse

Hello!
"react-native": "0.55.3",
"react-native-onesignal": "^3.2.3",

app.js
```
componentWillMount() {
    OneSignal.init('bddacd3f-7812-4a29-b038-.................');
    OneSignal.addEventListener('received', this.onReceived);
    OneSignal.addEventListener('opened', this.onOpened);
    OneSignal.addEventListener('ids', this.onIds);
}



onReceived(notification) {
        console.log("Notification received: ", notification);
    }

    onOpened(openResult) {
      console.log('Message: ', openResult.notification.payload.body);
      console.log('Data: ', openResult.notification.payload.additionalData);
      console.log('isActive: ', openResult.notification.isAppInFocus);
      console.log('openResult: ', openResult);
    }



    onIds(device) {
		console.log('Device info: ', device);
   }
```
1. work on iOS

2. not work on Android

I did
```
componentDidMount(){
    setTimeout(()=>{
      OneSignal.init('bddacd3f-7812-4a29-b038-ebf844c9826d');
    },5000);
}

```

```
E/OneSignal: The included Android Support Library is to old or incomplete. Please update to the 26.0.0 revision or newer.
D/FirebaseApp: com.google.firebase.auth.FirebaseAuth is not linked. Skipping initialization.
I/zygote: Rejecting re-init on previously-failed class java.lang.Class<com.onesignal.OneSignalChromeTab$OneSignalCustomTabsServiceConnection>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/customtabs/CustomTabsServiceConnection;
I/zygote:     at void com.onesignal.OneSignalChromeTab.setup(android.content.Context, java.lang.String, java.lang.String, java.lang.String) (OneSignalChromeTab.java:46)
I/zygote:     at void com.onesignal.OneSignal$7.run() (OneSignal.java:1164)
I/zygote:     at void java.lang.Thread.run() (Thread.java:764)
I/zygote: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.support.customtabs.CustomTabsServiceConnection" on path: DexPathList[[zip file "/data/app/com.publikun-tQvVeGt0XjzJehSUmUpCQw==/base.apk"],nativeLibraryDirectories=[/data/app/com.publikun-tQvVeGt0XjzJehSUmUpCQw==/lib/x86, /data/app/com.publikun-tQvVeGt0XjzJehSUmUpCQw==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
I/zygote:     at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
I/zygote:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
I/zygote:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
I/zygote:     at void com.onesignal.OneSignalChromeTab.setup(android.content.Context, java.lang.String, java.lang.String, java.lang.String) (OneSignalChromeTab.java:46)
I/zygote:     at void com.onesignal.OneSignal$7.run() (OneSignal.java:1164)
I/zygote:     at void java.lang.Thread.run() (Thread.java:764)
I/zygote: Rejecting re-init on previously-failed class java.lang.Class<com.onesignal.OneSignalChromeTab$OneSignalCustomTabsServiceConnection>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/customtabs/CustomTabsServiceConnection;
I/zygote:     at void com.onesignal.OneSignalChromeTab.setup(android.content.Context, java.lang.String, java.lang.String, java.lang.String) (OneSignalChromeTab.java:46)
I/zygote:     at void com.onesignal.OneSignal$7.run() (OneSignal.java:1164)
I/zygote:     at void java.lang.Thread.run() (Thread.java:764)
I/zygote: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.support.customtabs.CustomTabsServiceConnection" on path: DexPathList[[zip file "/data/app/com.publikun-tQvVeGt0XjzJehSUmUpCQw==/base.apk"],nativeLibraryDirectories=[/data/app/com.publikun-tQvVeGt0XjzJehSUmUpCQw==/lib/x86, /data/app/com.publikun-tQvVeGt0XjzJehSUmUpCQw==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
I/zygote:     at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
I/zygote:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
I/zygote:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
I/zygote:     at void com.onesignal.OneSignalChromeTab.setup(android.content.Context, java.lang.String, java.lang.String, java.lang.String) (OneSignalChromeTab.java:46)
I/zygote:     at void com.onesignal.OneSignal$7.run() (OneSignal.java:1164)
I/zygote:     at void java.lang.Thread.run() (Thread.java:764)
I/zygote: Rejecting re-init on previously-failed class java.lang.Class<com.onesignal.OneSignalChromeTab$OneSignalCustomTabsServiceConnection>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/customtabs/CustomTabsServiceConnection;
I/zygote:     at void com.onesignal.OneSignalChromeTab.setup(android.content.Context, java.lang.String, java.lang.String, java.lang.String) (OneSignalChromeTab.java:46)
I/zygote:     at void com.onesignal.OneSignal$7.run() (OneSignal.java:1164)
I/zygote:     at void java.lang.Thread.run() (Thread.java:764)
I/zygote: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.support.customtabs.CustomTabsServiceConnection" on path: DexPathList[[zip file "/data/app/com.publikun-tQvVeGt0XjzJehSUmUpCQw==/base.apk"],nativeLibraryDirectories=[/data/app/com.publikun-tQvVeGt0XjzJehSUmUpCQw==/lib/x86, /data/app/com.publikun-tQvVeGt0XjzJehSUmUpCQw==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
I/zygote:     at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
I/zygote:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
I/zygote:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
I/zygote:     at void com.onesignal.OneSignalChromeTab.setup(android.content.Context, java.lang.String, java.lang.String, java.lang.String) (OneSignalChromeTab.java:46)
I/zygote:     at void com.onesignal.OneSignal$7.run() (OneSignal.java:1164)
I/zygote:     at void java.lang.Thread.run() (Thread.java:764)

```

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -5,8 +5,8 @@ def safeExtGet(prop, fallback) {
 }
 
 android {
-    compileSdkVersion safeExtGet('compileSdkVersion', 23)
-    buildToolsVersion safeExtGet('buildToolsVersion', '23.0.1')
+    compileSdkVersion safeExtGet('compileSdkVersion', 26)
+    buildToolsVersion safeExtGet('buildToolsVersion', '26.0.2')
 
     defaultConfig {
         minSdkVersion safeExtGet('minSdkVersion', 16)

--- a/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
+++ b/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
@@ -75,7 +75,6 @@ public class RNOneSignal extends ReactContextBaseJavaModule implements Lifecycle
    // However it seems it is also to soon to call getCurrentActivity() from the reactContext as well.
    // This will normally succeed when onHostResume fires instead.
    private void initOneSignal() {
-
       // Uncomment to debug init issues.
       // OneSignal.setLogLevel(OneSignal.LOG_LEVEL.VERBOSE, OneSignal.LOG_LEVEL.ERROR);
 
@@ -107,8 +106,23 @@ public class RNOneSignal extends ReactContextBaseJavaModule implements Lifecycle
    @ReactMethod 
    public void init(String appId) {
       Activity activity = getCurrentActivity();
-      if (activity == null || oneSignalInitDone) {
-         Log.e("onesignal", "Unable to initialize the OneSignal SDK because activity is null " + (activity == null) + " or oneSignalInitDone" + oneSignalInitDone);
+      
+      if (activity == null) {
+         // in some cases, especially with react-native-navigation, it can take a while for the Activity to be created
+         // if null, we should re-attempt initialization after 50 milliseconds.
+         final String currentAppId = appId;
+         Timer timer = new Timer();
+         timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+               init(currentAppId);
+            }
+         }, 50);
+         return;
+      }
+
+      if (oneSignalInitDone) {
+         Log.w("onesignal", "The OneSignal SDK has already been initialized");
          return;
       }
 

--- a/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
+++ b/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
@@ -105,7 +105,7 @@ public class RNOneSignal extends ReactContextBaseJavaModule implements Lifecycle
 
    @ReactMethod 
    public void init(String appId) {
-      Activity activity = getCurrentActivity();
+      Activity activity = mReactApplicationContext.getCurrentActivity();
       
       if (activity == null) {
          // in some cases, especially with react-native-navigation, it can take a while for the Activity to be created

--- a/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
+++ b/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
@@ -106,37 +106,33 @@ public class RNOneSignal extends ReactContextBaseJavaModule implements Lifecycle
    @ReactMethod 
    public void init(String appId) {
       Activity activity = mReactApplicationContext.getCurrentActivity();
-      
-      if (activity == null) {
-         // in some cases, especially with react-native-navigation, it can take a while for the Activity to be created
-         // if null, we should re-attempt initialization after 50 milliseconds.
-         final String currentAppId = appId;
-         Timer timer = new Timer();
-         timer.schedule(new TimerTask() {
-            @Override
-            public void run() {
-               init(currentAppId);
-            }
-         }, 50);
-         return;
-      }
 
       if (oneSignalInitDone) {
-         Log.w("onesignal", "The OneSignal SDK has already been initialized");
+         Log.w("onesignal", "Already initialized the OneSignal React-Native SDK");
          return;
       }
 
       oneSignalInitDone = true;
-      
 
       OneSignal.sdkType = "react";
       
-      OneSignal.init(activity,
-         null,
-         appId,
-         new NotificationOpenedHandler(mReactContext),
-         new NotificationReceivedHandler(mReactContext)
-      );
+      if (activity == null) {
+         // in some cases, especially when react-native-navigation is installed,
+         // the activity can be null, so we can initialize with the context instead
+         OneSignal.init(mReactApplicationContext.getApplicationContext(),
+                 null,
+                 appId,
+                 new NotificationOpenedHandler(mReactContext),
+                 new NotificationReceivedHandler(mReactContext)
+         );
+      } else {
+         OneSignal.init(activity,
+                 null,
+                 appId,
+                 new NotificationOpenedHandler(mReactContext),
+                 new NotificationReceivedHandler(mReactContext)
+         );
+      }
    }
 
    @ReactMethod

--- a/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
+++ b/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java
@@ -105,10 +105,10 @@ public class RNOneSignal extends ReactContextBaseJavaModule implements Lifecycle
 
    @ReactMethod 
    public void init(String appId) {
-      Activity activity = mReactApplicationContext.getCurrentActivity();
+      Context context = mReactApplicationContext.getCurrentActivity();
 
       if (oneSignalInitDone) {
-         Log.w("onesignal", "Already initialized the OneSignal React-Native SDK");
+         Log.e("onesignal", "Already initialized the OneSignal React-Native SDK");
          return;
       }
 
@@ -116,23 +116,18 @@ public class RNOneSignal extends ReactContextBaseJavaModule implements Lifecycle
 
       OneSignal.sdkType = "react";
       
-      if (activity == null) {
+      if (context == null) {
          // in some cases, especially when react-native-navigation is installed,
          // the activity can be null, so we can initialize with the context instead
-         OneSignal.init(mReactApplicationContext.getApplicationContext(),
-                 null,
-                 appId,
-                 new NotificationOpenedHandler(mReactContext),
-                 new NotificationReceivedHandler(mReactContext)
-         );
-      } else {
-         OneSignal.init(activity,
-                 null,
-                 appId,
-                 new NotificationOpenedHandler(mReactContext),
-                 new NotificationReceivedHandler(mReactContext)
-         );
+         context = mReactApplicationContext.getApplicationContext();
       }
+
+      OneSignal.init(context,
+              null,
+              appId,
+              new NotificationOpenedHandler(mReactContext),
+              new NotificationReceivedHandler(mReactContext)
+      );
    }
 
    @ReactMethod

